### PR TITLE
Adds support for saving and editing an application

### PR DESCRIPTION
**Pull request by IAPark**
Adds an application manager class which is capable of saving and fetching an application from a server.

### CODE REVIEW

File: src/UserManager.ts [hunk 1]
```diff
@@ -4,6 +4,14 @@ import { ServiceClass } from './ServiceClass';
 import { AxiosInstance } from 'axios';
 import { AuthHelper } from './util/AuthenticationHelper'
 
+declare function require(name: any): any;
+
+// Node specific storage
+if (typeof localStorage === "undefined" || localStorage === null) {
+    var LocalStorage = require('node-localstorage').LocalStorage;
+    var localStorage = new LocalStorage('/tmp/test');
```
Comment (Brojowski) [inline src/UserManager.ts]: Do we want to put this somewhere other than test?

Comment (Brojowski) [inline src/UserManager.ts]: Could we rename the lowercase localstorage? Differing by one letter could be pretty confusing.

File: src/UserManager.ts [hunk 2]
```diff
@@ -30,16 +38,13 @@ export class UserManager extends ServiceClass {
...
         localStorage.setItem(UserManager.localStorageKey, JSON.stringify(this.currentUser));
         log.debug(['Saved User Locally', user])
     }
-
-    private setCurrentUser(user: User) {
-        
-    }
     
-    login(email: string, password: string): Promise<any>
+    login(email: string, password: string): Promise<User>
```
Comment (Brojowski) [inline src/UserManager.ts]: Thank you for fixing this for me.

File: src/UserManager.ts [hunk 3]
```diff
@@ -48,32 +53,25 @@ export class UserManager extends ServiceClass {
...
-        })).then((res) => {
-            _user.application = res.data.application;
-            log.debug([UserManager, 'Loaded User Application', _user.application])
-        }).catch((err) => {
-            throw err;
-        })
-    }
-
-    createUser(email: string, password:string): Promise<any> {
+    createUser(email: string, password: string): Promise<any> {
```
Comment (Brojowski) [inline src/UserManager.ts]: Do we want to make create user return the user from the promise to match login?

File: src/UserManager.ts (hunk 1 above)
Comment (IAPark) [reply src/UserManager.ts]: lower case localstorage is normally a standard part of the web browser API, but we could change LocalStorage to something else or just not even use it, though the line would be long

File: src/UserManager.ts (hunk 3 above)
Comment (IAPark) [reply src/UserManager.ts]: That sounds like a good idea

**Pull request merged**